### PR TITLE
update(staticmaps): add missing props to MapOptions

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- `subdomains`
- `tileRequestLimit`
- `zoomRange`

https://github.com/StephanGeorg/staticmaps#map-options

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present)
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## staticmaps ([<kbd>unpkg</kbd>](https://unpkg.com/browse/staticmaps@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'staticmaps'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4d9286d8d89285577bec872ec7ceee336d320317
</p>

File: types/staticmaps/index.d.ts [hunk 1]
```diff
@@ -26,15 +26,35 @@ declare class StaticMapsImage {
     buffer: (mime?: string, outputOptions?: OutputOptions | PngOptions | JpegOptions | WebpOptions) => Promise<Buffer>;
 }
 
+export type ZoomRange = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17;
```
Comment (OlivierKamers) [inline types/staticmaps/index.d.ts:29]: For our custom tiles we go up to 23. Is there a large advantage of using this union over the general `number` type?
The naming of the option `zoomRange` is to indicate a range (min and max) of zoomlevels so as a minor comment this type alias should be `ZoomLevel`

**Review by OlivierKamers — CHANGES_REQUESTED** [review]

File: types/staticmaps/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/staticmaps/index.d.ts]: > is there a large advantage of using this union over the general `number` type?

There are at least two, intellisense and keeping in the range, but that's disputable. I can amend this and just use number (and correct JSDoc). Just let me know

image
> Type '25' is not assignable to type '1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | undefined'.



> The naming of the option `zoomRange` is to indicate a range (min and max) of zoomlevels so as a minor comment this type alias should be `ZoomLevel`

amended, thanks!

**Review by OlivierKamers — APPROVED** [review]
Looks good 👍

File: types/staticmaps/index.d.ts (hunk 1 above)
Comment (OlivierKamers) [reply types/staticmaps/index.d.ts]: Okay I see, the union looks good 👍 Thanks!

**Comment by peterblazejewicz** [comment]
Ready to merge 💋

**Pull request merged**